wallsAndGates.js

```diff
@@ -0,0 +1,122 @@
+/*
+You are given an m x n grid rooms initialized with these three possible values.
+
+- -1 A wall or an obstacle.
+- 0 A gate.
+- INF Infinity means an empty room. We use the value 2^31 - 1 = 2147483647 to represent INF as you may assume that the distance to a gate is less than 2147483647.
+
+Fill each empty room with the distance to its nearest gate. If it is impossible to reach a gate, it should be filled with INF.
+*/
+
+const { Queue } = require('./utils/queue');
+
+const directions = [
+    [0, 1],
+    [1, 0],
+    [0, -1],
+    [-1, 0],
+];
+
+const GATE = 0;
+const WALL = -1;
+
+const isInBound = (matrix, row, col) => {
+    const rowBoundLength = matrix.length;
```

Comment:
`rowBoundLength` is a confusing name. I don't know what that means.

You should describe what real-world quantity this represents, you don't want to just describe what the code after the = does. In this case, `matrix.length` gives us the number of rows of the matrix. So something like `numRows` would be a clear name.